### ISSUE TITLE
Index out of Error in metrics.py

### ISSUE THREAD

**Issue by swaingotnochill** [ultralytics/yolov5#2766]
## ❔Question
I only want to detect a single class 'person'. I used 0 for its class id in the validation labels. When I used test.py, it didn't evaluate properly  and gave error Index out of error in metrics.py. Am I missing something, like some type of modification in the metrics file for single class detection or if there is anything Wrong in my approach, plzz tell me.

## Additional context
I am testing on custom dataset cityperson.

**Comment by glenn-jocher** [comment]
@swaingotnochill 👋 Hello, thank you for your interest in our work! This issue seems to lack the minimum requirements for a proper response, or is insufficiently detailed for us to help you. Please note that most technical problems are due to:

- **Your modified or out-of-date code**. If your issue is not reproducible in a new `git clone` version of this repo we can not debug it. Before going further run this code and verify your issue persists:
```bash
$ git clone https://github.com/ultralytics/yolov5 yolov5_new  # clone latest
$ cd yolov5_new
$ python detect.py  # verify detection

# CODE TO REPRODUCE YOUR ISSUE HERE
```

- **Your custom data.** If your issue is not reproducible in one of our 3 common datasets ([COCO](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml), [COCO128](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), or [VOC](https://github.com/ultralytics/yolov5/blob/master/data/voc.yaml)) we can not debug it. Visit our [Custom Training Tutorial](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data) for guidelines on training your custom data. Examine `train_batch0.jpg` and `test_batch0.jpg` for a sanity check of your labels and images.

- **Your environment.** If your issue is not reproducible in one of the verified environments below we can not debug it. If you are running YOLOv5 locally, verify your environment meets all of the [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies specified below. If in doubt, download Python 3.8.0 from https://www.python.org/, create a new [venv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/), and install requirements.

If none of these apply to you, we suggest you close this issue and raise a new one using the 🐛 **Bug Report template**, providing screenshots and a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) of your issue. Thank you!


## Requirements

Python 3.8 or later with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies installed, including `torch>=1.7`. To install run:
```bash
$ pip install -r requirements.txt
```


## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu.

**Comment by swaingotnochill** [comment]
@glenn-jocher Can you look into the screenshot and let me know where's the issue?
![Screenshot from 2021-04-13 14-51-47](https://user-images.githubusercontent.com/54025913/114529742-e60b5900-9c67-11eb-8d65-e4949c72bf63.png)

**Comment by swaingotnochill** [comment]
I also tried to recreate the same with coco128. You can also see the coco128.yaml file 
![Screenshot from 2021-04-13 15-27-26](https://user-images.githubusercontent.com/54025913/114534638-d93d3400-9c6c-11eb-93d6-fed34b08ce72.png)
@glenn-jocher

**Comment by glenn-jocher** [comment]
@swaingotnochill we can't help in debugging custom code or notebooks. I would recommend you start from the official notebook, which demos correct operation of all tasks (train, test, detect, export etc.)
https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb

**Comment by swaingotnochill** [comment]
How about a single class training example?

**Comment by glenn-jocher** [comment]
@swaingotnochill single-class training works the same as normal training. You can also optionally train any dataset in single-class mode with:
```
python train.py --single-cls
```
